synapseutils/sync.py

```diff
@@ -1021,9 +1130,50 @@ def syncToSynapse(
         _manifest_upload(syn, df)
 
 
-def _manifest_upload(syn, df):
+def _convert_cell_in_manifest_to_python_types(
+    cell: str,
+) -> typing.Union[typing.List, datetime.datetime, float, int, bool, str]:
+    """
+    Takes a possibly semi-colon delimited cell from the manifest TSV file into a list
+    of items to be used as annotations.
+
+    Args:
+        cell: The cell item to convert.
+
+    Returns:
+        The list of items to be used as annotations. Or a single instance if that is
+            all that is present.
+    """
+    values_to_return = []
+
+    cell_values = cell.split(";")
+    for annotation_value in cell_values:
+        if possible_datetime := datetime_or_none(annotation_value):
```

Comment:
Yeah I can add it here. Specifically I added it below because a bool `False` and number `0` would cause the if check to fail. In this datetime case we are getting back an Object or None.